packages/ozone/src/mod-service/views.ts

```diff
@@ -408,7 +417,28 @@ export class ModerationViews {
       .if(!includeNeg, (qb) => qb.where('neg', '=', false))
       .selectAll()
       .execute()
-    return res.map((l) => formatLabel(l))
+    return Promise.all(res.map((l) => this.formatLabelAndEnsureSig(l)))
+  }
+
+  async formatLabelAndEnsureSig(row: LabelRow) {
+    const signingKey = this.signingKey.did()
+    const formatted = formatLabel(row)
+    if (!!row.sig && row.signingKey === signingKey) {
+      return formatted
+    }
+    const signed = await signLabel(formatted, this.signingKey)
+    this.backgroundQueue.add(async (db) => {
+      try {
+        await db.db
+          .updateTable('label')
+          .set({ sig: Buffer.from(signed.sig), signingKey })
+          .where('id', '=', row.id)
+          .execute()
+      } catch (err) {
+        dbLogger.error({ err, label: row }, 'failed to update resigned label')
+      }
+    })
```

Comment:
If someone resyncs all labels after changing your signing key, things could get a bit interesting here.  If that's a concern, I think the most at-hand thing to do would be to introduce some batching.

Reply:
actually yeah i like the idea of just doing it inline for back pressure - we'll still catch/log the error & send out the signed label if this fails for some reason